File: python/robot-return-to-origin.py
OVES = {'U': 1, 'D': -1, "L": 100000, "R": -100000}

class Solution:
    def judgeCircle(self, moves: str) -> bool:
        x = 0
        for c in moves:
            x += MOVES[c]
        return x == 0

File: rust/robot-return-to-origin.rs
impl Solution {
    pub fn judge_circle(moves: String) -> bool {
        let mut x:i32 = 0;
        for c in moves.chars() {
            match c {
                'U' => x += 1,
                'D' => x -= 1,
                'L' => x += 100000,
                _ => x -= 100000
            }
        }
        return x == 0;
    }
}
